pkg/cmd/objects/update/object.go

```diff
@@ -1,78 +1,78 @@
-package updateObjects
+package update
 
 import (
 	"encoding/json"
 	"fmt"
 
 	"github.com/algolia/algoliasearch-client-go/v3/algolia/search"
 	"github.com/algolia/cli/pkg/utils"
+	"github.com/mitchellh/mapstructure"
 )
 
-type ObjectsToUpdate []map[string]interface{}
+// Object is a map[string]interface{} that can be unmarshalled from a JSON object
+// The object must have an objectID field
+// Each field could be either an `search.PartialUpdateOperation` or a scalar value
+type Object map[string]interface{}
 
-func (o *ObjectsToUpdate) UnmarshalJSON(data []byte) error {
-	var rawObjectsToUpdate []map[string]interface{}
-	err := json.Unmarshal(data, &rawObjectsToUpdate)
-	if err != nil {
-		return err
-	}
+// Valid operations
+const (
+	Increment     string = "Increment"
+	Decrement     string = "Decrement"
+	Add           string = "Add"
+	AddUnique     string = "AddUnique"
+	IncrementSet  string = "IncrementSet"
+	IncrementFrom string = "IncrementFrom"
+)
+
+// ValidateOperation checks that the operation is valid
+func ValidateOperation(p search.PartialUpdateOperation) error {
```

Comment:
Should we create a unit test for that function?

Reply:
Yep 👍